schema.sql

```diff
@@ -0,0 +1,74 @@
+
+CREATE TABLE country (
+    country_id BIGINT PRIMARY KEY,
+    country_name VARCHAR UNIQUE NOT NULL
+);
+
+
+CREATE TABLE customer (
+    customer_id BIGINT PRIMARY KEY,
+    customer_name VARCHAR NOT NULL,
+    country_id BIGINT,
+    FOREIGN KEY (country_id) REFERENCES country(country_id)
+);
+
+
+CREATE TABLE artist (
+    artist_id BIGINT PRIMARY KEY,
+    artist_name VARCHAR NOT NULL,
+    country_id BIGINT,
+    FOREIGN KEY (country_id) REFERENCES country(country_id)
+);
+
+
+CREATE TABLE genre (
+    genre_id BIGINT PRIMARY KEY,
+    genre_name VARCHAR NOT NULL
+);
+
+
+CREATE TABLE type (
+    type_id BIGINT PRIMARY KEY,
+    type_name VARCHAR NOT NULL
+);
+
+
+CREATE TABLE release (
+    release_id BIGINT PRIMARY KEY,
+    release_name VARCHAR NOT NULL,
+    release_date DATE NOT NULL,
+    genre_id BIGINT,
+    artist_id BIGINT,
+    type_id BIGINT,
+    FOREIGN KEY (genre_id) REFERENCES genre(genre_id),
+    FOREIGN KEY (artist_id) REFERENCES artist(artist_id),
+    FOREIGN KEY (type_id) REFERENCES type(type_id)
+);
+
+
+CREATE TABLE sale (
+    sale_id BIGINT PRIMARY KEY,
+    sale_price FLOAT NOT NULL,
+    sale_date DATE NOT NULL,
+    customer_id BIGINT,
+    release_id BIGINT,
+    FOREIGN KEY (customer_id) REFERENCES customer(customer_id),
+    FOREIGN KEY (release_id) REFERENCES release(release_id)
+);
+
+
+CREATE TABLE subscriber (
+    subscriber_id BIGINT PRIMARY KEY,
+    subscriber_email VARCHAR UNIQUE NOT NULL,
+    alerts BOOLEAN DEFAULT FALSE,
+    CHECK (subscriber_email ~* '^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}$')
```

Comment:
Is it possible to use the full regex for emails here:
```bash
^((([!#$%&'*+\-/=?^_`{|}~\w])|([!#$%&'*+\-/=?^_`{|}~\w][!#$%&'*+\-/=?^_`{|}~\.\w]{0,}[!#$%&'*+\-/=?^_`{|}~\w]))[@]\w+([-.]\w+)*\.\w+([-.]\w+)*)$
```

Reply:
According to google, when working with SQL it's better to use a simpler regex. Apparently it's better suited for database constraints and overly strict regex can lead to false negatives, rejecting emails that are valid but uncommon